lib/charms/opensearch/v0/opensearch_locking.py

```diff
@@ -266,10 +284,10 @@ def acquired(self) -> bool:  # noqa: C901
                 # if the node lock cannot be acquired, fall back to peer databag lock
                 # this avoids hitting deadlock situations in cases where
                 # the .charm_node_lock index is not available
-                if online_nodes <= 1:
+                if online_nodes <= 1 and self._charm.app.planned_units() > 1:
                     return self._peer.acquired
                 else:
-                    return False
+                    return self._charm.app.planned_units() > 1
```

Comment:
I'm not sure if this is correct: If there's an error when querying which unit has the lock, and there's more than one unit planned, you just want to return true? If several units do this at once, they'd all just go down (or start) at once.